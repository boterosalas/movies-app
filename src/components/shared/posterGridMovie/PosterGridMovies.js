import React from 'react'
import { Link } from 'react-router-dom';
import { getPosterUrl } from '../../helpers/helpers';
import './PosterGridMovies.css';

const PosterGridMovies = ({ moviesList }) => {

    return (
        <div className="row">
            {
                moviesList.map(movie => {
                    return (
                        <div key={movie.id} className="col-6 col-sm-4 col-md-3 mb-4">
                            <Link to={`/movie/${movie.id}`}>
                                <img src={getPosterUrl(movie.poster_path)} alt={movie.title} className="img-fluid poster" />
                            </Link>
                            <div>
                                <p className="text-warning text-right m-0 w-100">
                                    {movie.vote_average}
                                </p>
                                <h3 className="movie-title">{movie.title}</h3>
                            </div>
                        </div>
                    )
                })
            }
        </div>
    )
}

export default PosterGridMovies
