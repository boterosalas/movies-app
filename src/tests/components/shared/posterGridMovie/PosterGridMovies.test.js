import React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import PosterGridMovies from '../../../../components/shared/posterGridMovie/PosterGridMovies';
import { MemoryRouter } from 'react-router-dom';


describe('Test on <PosterGridMovies />', () => {

    const moviesList = [
        {
            id: 1,
            poster_path: '',
            title: '',
            vote_average: ''
        },
        {
            id: 2,
            poster_path: '',
            title: '',
            vote_average: ''
        },
        {
            id: 3,
            poster_path: '',
            title: '',
            vote_average: ''
        },
    ]

    const wrapper = mount(
        <MemoryRouter>
            <PosterGridMovies moviesList={moviesList} />
        </MemoryRouter>
    );

    test('Should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Link').length).toBe(moviesList.length);
        expect(wrapper.find('img').length).toBe(moviesList.length);
        expect(wrapper.find('.movie-title').length).toBe(moviesList.length);
    });

    test('Should coincide the title elements with the text on title tags', () => {
        wrapper.find('.movie-title').forEach((title, idx) => {
            expect(title.text()).toBe(moviesList[idx].title);
        });
    });


})