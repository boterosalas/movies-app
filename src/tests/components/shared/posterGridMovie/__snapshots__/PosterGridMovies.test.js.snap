// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <PosterGridMovies /> Should display the component correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-6 col-sm-4 col-md-3 mb-4"
    key="1"
  >
    <a
      href="/movie/1"
      onClick={[Function]}
    >
      <img
        alt=""
        className="img-fluid poster"
        src="/assets/no-image.jpg"
      />
    </a>
    <div>
      <p
        className="text-warning text-right m-0 w-100"
      />
      <h3
        className="movie-title"
      />
    </div>
  </div>
  <div
    className="col-6 col-sm-4 col-md-3 mb-4"
    key="2"
  >
    <a
      href="/movie/2"
      onClick={[Function]}
    >
      <img
        alt=""
        className="img-fluid poster"
        src="/assets/no-image.jpg"
      />
    </a>
    <div>
      <p
        className="text-warning text-right m-0 w-100"
      />
      <h3
        className="movie-title"
      />
    </div>
  </div>
  <div
    className="col-6 col-sm-4 col-md-3 mb-4"
    key="3"
  >
    <a
      href="/movie/3"
      onClick={[Function]}
    >
      <img
        alt=""
        className="img-fluid poster"
        src="/assets/no-image.jpg"
      />
    </a>
    <div>
      <p
        className="text-warning text-right m-0 w-100"
      />
      <h3
        className="movie-title"
      />
    </div>
  </div>
</div>
`;
